engine/src/main/java/io/camunda/zeebe/streamprocessor/ProcessingStateMachine.java

```diff
@@ -327,8 +325,6 @@ private void errorHandlingInTransaction(final Throwable processingException) thr
           final ProcessingResultBuilder processingResultBuilder =
               new DirectProcessingResultBuilder(
                   context, logStreamBatchWriter::canWriteAdditionalEvent);
-          // todo(#10047): replace this reset method by using Buffered Writers
-          processingResultBuilder.reset();
```

Comment:
> will use a new ProcessingResultBuilder

❓ I expect that we can only get rid of this reset when we remove the `DirectProcessingResultBuilder` and replace it with a buffered version. However, that class still exists. In addition, using a new builder for the command and the error handling doesn't resolve this problem because the underlying structure must still be reset. How come we can remove this reset now?

Reply:
We always use a new builder, and this builder uses an "buffered" approach with the RecordBatch classes. There is no longer any direct write to a dispatcher etc. 

The name of the ResultBuilder can be changed.

Reply:
Ah okay I guess the name is confusing. Please have a look at that, then I'm good with this change 👍

Reply:
Follow up https://github.com/camunda/zeebe/issues/10216